required/latex-lab/latex-lab-table.dtx

```diff
@@ -1015,26 +1049,58 @@
  }
 %    \end{macrocode}
 %  \end{macro}
+%
+% \begin{macro}{\@@_init_tagnames_default:,\@@_init_tagnames_presentation:} 
+% These commands are used to switch the tagnames. 
+%    \begin{macrocode}
+\cs_new_protected:Npn\@@_init_tagnames_default:
+  {
+    \tl_set:Nn\l_@@_rowtag_tl   {TR}
+    \tl_set:Nn\l_@@_pcelltag_tl {TD}
+    \tl_set:Nn\l_@@_celltag_tl  {TD}
+    \tl_set:Nn\l_@@_tabletag_tl {Table} 
+  }     
+
+\cs_new_protected:Npn\@@_init_tagnames_presentation:
+  {
+    \tl_set:Nn\l_@@_rowtag_tl   {NonStruct}
+    \tl_set:Nn\l_@@_pcelltag_tl {NonStruct}
+    \tl_set:Nn\l_@@_celltag_tl  {text}
+    \tl_set:Nn\l_@@_tabletag_tl {Div} 
+  }     
+%    \end{macrocode}
+% \end{macro}
 
 %    \begin{macrocode}
 % See tagpdfsetup-keys.md in tagpdf/doc for the naming scheme.
 \keys_define:nn { __tag / setup }
   {
     table/tagging .choices:nn = { true, on }
-      { \__tag_tbl_enable: },
+      { 
+        \__tag_tbl_enable: 
+        \@@_init_tagnames_default:
+      },
     table/tagging .choices:nn = { false, off }
       { \__tag_tbl_disable: },
     table/tagging .choice:,
     table/tagging / presentation .code:n = 
-     {
-       \__tag_tbl_enable:
-       \tl_set:Nn\l_@@_rowtag_tl   {NonStruct}
-       \tl_set:Nn\l_@@_pcelltag_tl {NonStruct}
-       \tl_set:Nn\l_@@_celltag_tl  {text}
-       \tl_set:Nn\l_@@_tabletag_tl {Div} 
-       \clist_clear:N \l_@@_header_rows_clist
-       \clist_clear:N \l_@@_header_columns_clist  
-     },  
+      {
+        \__tag_tbl_enable:
+        \@@_init_tagnames_default:
+        \AssignSocketPlug{tagsupport/tbl/hmode/begin}{LayoutTable}
+        \AssignSocketPlug{tagsupport/tbl/vmode/begin}{LayoutTable}
+        \clist_clear:N \l_@@_header_rows_clist
+        \clist_clear:N \l_@@_header_columns_clist  
+      },  
+    table/tagging / div .code:n = 
+      {
+        \__tag_tbl_enable:
+        \@@_init_tagnames_presentation:
+        \AssignSocketPlug{tagsupport/tbl/hmode/begin}{LayoutTable}
+        \AssignSocketPlug{tagsupport/tbl/vmode/begin}{LayoutTable}
+        \clist_clear:N \l_@@_header_rows_clist
+        \clist_clear:N \l_@@_header_columns_clist  
```

Comment:
am I missing something? This looks 100% like the previous case

Reply:
the difference is `\@@_init_tagnames_default:` and `\@@_init_tagnames_presentation:`.

Reply:
hmm, yes, overlooked that, but isn't that a bit odd then? I mean in case of DIV it is called _presentation and in the other case it is called _default? I would expect the names and the case to match

Reply:
hm right, the name was from an earlier version where presentation mapped to div, I will change the name to `\@@_init_tagnames_div:`.